yrs/src/update.rs

```diff
@@ -944,7 +944,8 @@ impl BlockCarrier {
             }
             BlockCarrier::Skip(x) => {
                 encoder.write_info(BLOCK_SKIP_REF_NUMBER);
-                encoder.write_len(x.len - offset);
+                // write as VarUint because Skips can't make use of predictable length-encoding
+                encoder.write_var(x.len - offset);
```

Comment:
I've tested your example, and as it turns out this line change alone is enough to make it pass - the change above doesn't seem to affect results. I've also prepared artificial test with skip blocks inside and it also passed with only this line present.